instrumentation/jmx-metrics/javaagent/src/main/resources/jmx/rules/camel.yaml

```
@@ -0,0 +1,263 @@
+      ExternalRedeliveries:
+        metric: exchange.redelivered.external
+        type: counter
+        unit: "{exchanges}"
+        desc: The total number of all external initiated redeliveries (such as from JMS broker) since processor start-up or the last reset operation.
+      Redeliveries:
+        metric: exchange.redelivered
+        type: counter
+        unit: "{exchanges}"
+        desc: Number of exchanges redelivered (internal only)  since selected processor start-up or the last reset operation.
+      MaxProcessingTime:
+        metric: exchange.processing.max_time
+        unit: ms
+        type: gauge
+        desc: Indicates the longest time, in milliseconds, to process an exchange since processor start-up or the last reset operation.
+      MeanProcessingTime:
+        metric: exchange.processing.mean_time
+        type: gauge
+        unit: ms
+        desc: Indicates the mean processing time, in milliseconds, for all exchanges processed since processor start-up or the last reset operation.
+      MinProcessingTime:
+        metric: exchange.processing.min_time
+        type: gauge
+        unit: ms
+        desc: Indicates the shortest time, in milliseconds, to process an exchange since processor start-up or the last reset operation.
+      LastProcessingTime:
+        metric: exchange.processing.last_time
+        type: gauge
+        unit: ms
+        desc: Indicates the time, in milliseconds, it took the selected processor to process the last exchange.
+      DeltaProcessingTime:
+        metric: exchange.processing.delta_time
+        type: gauge
+        unit: ms
+        desc: Indicates the difference, in milliseconds, of the Processing Time of the last two exchanges transited the selected processor.
+      TotalProcessingTime:
+        metric: exchange.processing.time
+        type: counter
+        unit: ms
+        desc: Indicates the total processing time, in milliseconds, to process all exchanges since start-up or the last reset operation.
+
+
+
+  - bean: org.apache.camel:context=*,type=threadpools,name=*
+    prefix: camel.threadpool.
+    metricAttribute:
+      route: beanattr(RouteId)
+      context: param(context)
+    mapping:
+      ActiveCount:
+        metric: active.count
+        type: updowncounter
+        unit: "{threads}"
+        desc: The approximate number of threads that are actively executing tasks.
+      CompletedTaskCount:
+        metric: completed.task.count
+        type: counter
+        unit: "{tasks}"
+        desc: The approximate total number of tasks that have completed execution. Because the states of tasks and threads may change dynamically during computation, the returned value is only an approximation, but one that does not ever decrease across successive calls.
+      PoolSize:
+        metric: pool.size
+        type: updowncounter
+        unit: "{threads}"
+        desc: The current number of threads in the pool.
+      LargestPoolSize:
+        metric: largest.pool.size
```

(similar to `exchange.processing.*_time`)
```suggestion
        metric: pool.largest_size
```